soc/arm/nxp_imx/rt5xx/Kconfig.defconfig.series

```diff
@@ -17,13 +17,29 @@ config NUM_IRQS
 config ZTEST_NO_YIELD
 	default y if (PM && ZTEST)
 
-#
-# The base address of the external flash comes from the FLEXSPI base
-# address. The size of the flash is defined by what is populated and
-# described in the board devicetree file.
-#
+# The base address is determined from the zephyr,flash node with the following
+# precedence:
+# FlexSPI base address (if flash node is on a FlexSPI bus)
+# node reg property (used for memory regions such as SRAM)
+
+# Workaround for not being able to have commas in macro arguments
+
+DT_CHOSEN_Z_FLASH := zephyr,flash
+DT_COMPAT_FLEXSPI := nxp,imx-flexspi
+
+# Macros to shorten Kconfig definitions
+DT_CHOSEN_FLASH_NODE := $(dt_chosen_path,$(DT_CHOSEN_Z_FLASH))
+DT_CHOSEN_FLASH_PARENT := $(dt_node_parent,$(DT_CHOSEN_FLASH_NODE))
+
 config FLASH_BASE_ADDRESS
-	default $(dt_node_reg_addr_hex,/soc/spi@134000,1)
+	default $(dt_node_reg_addr_hex,$(DT_CHOSEN_FLASH_PARENT),1) \
+		if $(dt_node_has_compat,$(DT_CHOSEN_FLASH_PARENT),$(DT_COMPAT_FLEXSPI))
+	default $(dt_chosen_reg_addr_hex,$(DT_CHOSEN_Z_FLASH))
+
+# Boards must add another default to this Kconfig to indicate the size of their
+# external flash, since the RT5xx has no internal flash.
+config FLASH_SIZE
+	default $(dt_chosen_reg_size_int,$(DT_CHOSEN_Z_FLASH),0,K)
```

Comment:
Why do we need set this board level. Can we get this from the zephyr,flash binding

Reply:
Sure, we can move the FLASH_SIZE setting to SOC level. I'll update this PR.